spark-confluence/src/main/java/com/k15t/spark/confluence/ConfluenceIframeSpaceAppAction.java

```diff
@@ -0,0 +1,93 @@
+package com.k15t.spark.confluence;
+
+import com.atlassian.confluence.spaces.Space;
+import com.atlassian.confluence.util.velocity.VelocityUtils;
+import com.k15t.spark.base.util.DocumentOutputUtil;
+import com.opensymphony.webwork.ServletActionContext;
+import org.jsoup.nodes.Document;
+
+import java.io.IOException;
+import java.util.Map;
+
+
+/**
+ * Class that can be extended for creating an action that opens a SPA in an iframe in the Space Tools view.
+ */
+public class ConfluenceIframeSpaceAppAction extends ConfluenceSpaceAppAction {
```

Comment:
This class should probably not extend the ```ConfluenceSpaceAppAction```, as it requires very little of the logic in there.

Instead it should extend Confluence's ```AbstractSpaceAction``` and implement ```SpaceAware```.

The selected web item should be retrieved in the same way as in ```ConfluenceSpaceAppAction```.
The title should be retrieved by subclasses implementing ```getTitle()```
The construction of the body is completely different anyway.

Changing this would also allow us to drop ```<param name="resource-path">``` from the action declaration in ```atlassian-plugin.xml```.